File: Android/app/src/main/java/com/edu/primary/database/entity/UserEntity.java
package com.edu.primary.database.entity;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "users")
public class UserEntity {
    @PrimaryKey(autoGenerate = true)
    public long id;

    public String username;
    public String password;
    public String nickname;
    public long createTime;

    public UserEntity() {
    }

    @Ignore
    public UserEntity(String username, String password, String nickname) {
        this.username = username;
        this.password = password;
        this.nickname = nickname;
        this.createTime = System.currentTimeMillis();
    }
}

File: Android-1/app/src/main/java/com/edu/primary/data/local/entity/UserEntity.kt
package com.edu.primary.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "users")
data class UserEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val username: String,
    val password: String,
    val nickname: String,
    val createTime: Long = System.currentTimeMillis()
)
